### ISSUE TITLE
User menu is not keyboard accessible

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-frontend#3110]
The dropdown user menu needs to be keyboard accessible otherwise those links are completely obscured to anyone that can't use a mouse.

The username menu should be focusable, and when focused that should open the menu and make it possible to tab down through the links.

<img width="405" alt="add-ons_for_firefox" src="https://user-images.githubusercontent.com/1514/30370757-6789c12e-9870-11e7-9c66-f31dcb4151f4.png">

**Comment by muffinresearch** [comment]
@willdurand pointed out on irc that enter does open the drop-down so this might be more a question of adding a focus style to the user button. 

Would also be worth reviewing if there's any additional hinting we should be doing via aria attrs to provide additional hints to assistive tech.

**Comment by willdurand** [comment]
On Chrome, this is what I observe when I use tab:

![2017-09-13 12 06 34](https://user-images.githubusercontent.com/217628/30371852-2cd861dc-987c-11e7-98db-bf7104b56b85.gif)

**Comment by muffinresearch** [comment]
IIRC chrome adds a lot in terms of default focus styles - we may just need an explicit focus style added to FF (Also I'd be curious to know if we're turning any defaults off for FF in CSS that we shouldn't be). If you're using chrome to develop it would be recommended to switch to nightly to make sure you're seeing things the way a firefox user would since that's our primary audience.

**Comment by willdurand** [comment]
Chrome adds a browser style for `*:focus`, which uses `outline`. This is also what Bootstrap does. An easy fix would be to add such a rule to have a consistent focus style on all browsers.

**Comment by willdurand** [comment]
> If you're using chrome to develop it would be recommended to switch to Firefox nightly to make sure you're seeing things the way a firefox user would since that's our primary audience.

I use Chrome, FF55 and FF Nightly.

**Comment by muffinresearch** [comment]
I've looked at this a bit more closely and the weird part about it, is the span seems to get focus after the button. 

Adding an explicit `:focus` style to the button means you can see when the button is focused but as the span is focusable (no idea why) you also need to work around the span focus by adding `tabindex="-1"` so that after the button the search box (or the links in the menu if the menu was opened are the next things to focus on).

**Comment by muffinresearch** [comment]
QA notes:

Before the fix - you can't see the focus on the user button when tabbing to it when logged in. Also the span around the username can be focused (1 tab after focusing the button) which doesn't make any sense.

With the fix in play:

* First log in
* Use the keyboard to tab to the user button
* Pressing space or enter opens the menu (this was already the case)
* Pressing tab again focuses on the links inside the menu.

<img width="461" alt="add-ons_for_firefox" src="https://user-images.githubusercontent.com/1514/30636511-93dadd30-9ded-11e7-8eb7-f688762095f9.png">


With the menu closed:

* Use the keyboard to tab to the user button
* Without opening the dropdown, pressing tab again should focus the search input.

<img width="352" alt="add-ons_for_firefox" src="https://user-images.githubusercontent.com/1514/30636534-aa8c9cf8-9ded-11e7-91ac-cfb5f66d1627.png">

I'm going to file another bug to make the search input focus styles more noticeable.

**Comment by muffinresearch** [comment]
I've filed a separate bug for the aria attributes: https://github.com/mozilla/addons-frontend/issues/3183

**Comment by vcarciu** [comment]
@muffinresearch : "More popular extensions", "Browse this collection", "More featured Themes" and "More trending extensions" are not focused while navigating with the keyboard. 
Also, the static panes which replaced the carousel are not highlighted when navigating with keyboard.

Please let me know if you want new issues or if we can track this here.

**Comment by vcarciu** [comment]
Verified as fixed. Logged another issue for the previous comment : #3568
